### PR TITLE
Use movie_id in place of movie_name in cache

### PR DESCRIPTION
**Pull request by SN-18**
This PR addresses issue #26
There are two functions that are added in cache.py in this pull request. One is to populate Redis for movies based on movie_id  and the other is to return results from Redis cache if movie_id is passed. This is essential because populating cache on movie_names could corrupt the cache. Example- There are two movies named "The Avengers", one from 1998 and the other from 2008. If cache result is populated based on movie_name alone, the cache would be poisoned and wrong results could be displayed when results from cache are returned. 
However, movie_id is unique and search based on movie_id would not corrupt the cache with wrong results.

### CODE REVIEW

File: src/cache.py [hunk 1]
```diff
@@ -23,6 +30,15 @@ def put_recommendation_for_movie_name(
...
+        self, movie_id: int, recommendation: List[Movie]
+    ):
+        list_of_movies = []
+        for movie in recommendation:
+            list_of_movies.append(movie.to_string())
+
+        self.connection.lpush(movie_id, *list_of_movies)
+
 
 if __name__ == "__main__":
```
Comment (TommasU) [inline src/cache.py:43]: This was supposed to be removed by Parth during testing. If we merge as we we need to resolve conflicts when the testing PR comes through.

**Review by TommasU — APPROVED** [review]
Will resolve minor conflicts when merging with Developer.

Methods look good otherwise.

File: src/cache.py (hunk 1 above)
Comment (pncnmnp) [reply src/cache.py:43]: I had removed it before my testing commit, but will do it again.

**Review by pncnmnp — APPROVED** [review]

**Review by rdeodha — APPROVED** [review]
Looks good

**Pull request merged**